_posts/2017-05-10-fastlane-travis-weekly-deploys.md

```diff
@@ -0,0 +1,249 @@
+
+My initial thoughts were to use a submodule, but that option provides little advantage over cloning the repo itself so it's done inline. Our dependencies for the app live in Rubygems (Fastlane/CocoaPods), NPM (React Native) and CocoaPods (Artsy Mobile code), so I use the `before_install` and `before_script` section of the `.travis.yml` to set up our dependencies:
+
+```yml
+# Use a Mac build please
+language: objective-c
+
+# Ensure our dependency managers are up to date
+before_install:
+  - brew install yarn
+  - gem install bundler
+  - bundle update
+
+# Clone emission and setup dependencies
+before_script:
+  - git clone https://github.com/artsy/emission.git
+  - cd emission && yarn install
+  - git pull origin fastlane_match 
+  - cd Example && pod install
+  - cd ../..
+```
+
+Note the `- bundle update`. As both CocoaPods and Fastlane are build tools that have to handle ever changing Apple black-boxes, then locking their versions doesn't make much sense over long time periods for somehting like this. In this case, both of the apps will always be at the latest stable release.
+
+## Ensuring signing will work.
+
+This one is a bit tricker, luckily I've already set up one of our apps to use [fastlane match][] and I can re-use that infrastructure. As it is a closed repo, Travis did not have access to clone the repo. I fixed this by creating an access token for a user with read-only access to our match-codesigning repo, then exposed this as a private environment variable in CI which the Matchfile uses. E.g.
+
+```ruby
+git_url "https://#{ENV['GITHUB_SUBMODULES_USER']}@github.com/artsy/mobile_fastlane_match"
+
+# Instead of 
+# git_url "https://github.com/artsy/mobile_fastlane_match"
+```
+
+That's what's great about building a DSL that which sits above a real programming language, you give users a lot of flexibility. I added a two fastlane lanes for code signing:
+
+```ruby
+# In case you need to update the signing profiles for this app
+lane :update_signing do
+  match(type: 'appstore')
+end
+
+# Used by CI, will not sneakily update (the CI only has read-only access to the repo anyway)
+lane :setup_signing do
+  match(type: 'appstore', readonly: true)
+end
+```
+
+## Creating the build and shipping it to Testflight
+
+This is handled by [fastlane gym][gym], again, I made a separate fastlane lane so it could be tested independently.
+
+```ruby
+# You can run `bundle exec fastlane build` to test the build process locally.
+lane :build do
+  gym workspace: 'emission/Example/Emission.xcworkspace',
+      configuration: 'Deploy',
+      scheme: 'Emission',
+      clean: true
+end
+```
+
+It uses a scheme for deploys, which prioritises using AppHub over a local React Native server. Gym handles a lot of CLI ugliness for us, and works well.
+
+Sending the app to Testflight involves a little bit of work:
+
+```ruby
+# Get the last 10 lines of the CHANGELOG for Testflight
+changelog = 'emission/CHANGELOG.md'
```

Comment:
There is a _fastlane_ plugin for that, maybe you can make use of it? https://github.com/pajapro/fastlane-plugin-changelog